import "galaxy.proto";
package baidu.galaxy;

option cc_generic_services = true;

message TaskInfoPersistence {
    optional string task_id = 1;
    optional TaskDescriptor task_desc = 2;
}

enum TaskStage {
    kTaskStagePENDING = 0;
    kTaskStageDEPLOYING = 1;
    kTaskStageRUNNING = 2;
    kTaskStageSTOPPING = 3;
    kTaskStageENDING = 4;
}

message PodInfoPersistence {
    optional string pod_id = 1;
    optional string job_id = 2;
    optional PodDescriptor pod_desc = 3;
    optional int32 initd_port = 5;
    optional int32 initd_pid = 6;
    repeated TaskInfoPersistence tasks = 7;
}

message QueryRequest {
}

message QueryResponse {
    optional Status status = 1;
    optional AgentInfo agent = 2;
}

message RunPodRequest {
    optional string podid = 1;
    optional PodDescriptor pod = 2;
    optional string jobid = 3;
}

message RunPodResponse {
    optional Status status = 1;
}

message KillPodRequest {
    optional string podid = 1;
}

message KillPodResponse {
    optional Status status = 1;
}

service Agent {
    rpc Query(QueryRequest) returns (QueryResponse);
    rpc RunPod(RunPodRequest) returns (RunPodResponse);
    rpc KillPod(KillPodRequest) returns (KillPodResponse);
}
